api/app/signals/apps/questionnaires/models/utils.py

```diff
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 Gemeente Amsterdam
+import os
+
+import matplotlib.pyplot as plt
+import networkx
+from django.utils import timezone
+from networkx.drawing.nx_pydot import graphviz_layout
+
+import pydot  # noqa
+
+
+def make_nx_graph(question_graph):
+    nx_graph = networkx.DiGraph()
+    nx_graph.add_node(question_graph.first_question_id)
+    for edge in question_graph.edges.all():
+        if edge.choice:
+            choice_label = f'{edge.choice.display or edge.choice.payload}' \
+                           f'{" (selected)" if edge.choice.selected else ""}'
+        else:
+            choice_label = ''
+
+        nx_graph.add_edge(edge.question_id, edge.next_question_id, choice_label=choice_label)
+
+    return nx_graph
+
+
+def validate_nx_graph(question_graph):
+    nx_graph = make_nx_graph(question_graph)
```

Comment:
This may end up using a new service - exact calling conventions of which are still TBD. See the (still evolving) PR 861.